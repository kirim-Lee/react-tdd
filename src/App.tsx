import { useState } from 'react';
import '../public/scss/style.scss';
import { Container, Contents, InputContainer, TodoList } from './Components';
import { TodoListProvider } from './Contexts/TodoList';

const App = () => {
  return (
    <TodoListProvider>
      <Container>
        <Contents>
          <TodoList />
          <InputContainer />
        </Contents>
      </Container>
    </TodoListProvider>
  );
};

export default App;
