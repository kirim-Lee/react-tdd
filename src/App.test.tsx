import { fireEvent, render, screen } from '@testing-library/react';
import App from './App';

describe('<App />', () => {
  it('렌더링이 잘 되는지 테스트', () => {
    const { container } = render(<App />);

    const toDoList = screen.getByTestId('toDoList');
    expect(toDoList).toBeInTheDocument();
    expect(toDoList.firstChild).toBeNull();

    const input = screen.getByPlaceholderText('할 일을 입력해 주세요');
    expect(input).toBeInTheDocument();
    const button = screen.getByText('추가');
    expect(button).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it('투두 아이템 추가 및 삭제 테스트', () => {
    render(<App />);

    const input = screen.getByPlaceholderText('할 일을 입력해 주세요');
    const button = screen.getByText('추가');
    fireEvent.change(input, { target: { value: 'study react 1' } });
    fireEvent.click(button);

    const todoItem = screen.getByText('study react 1');
    expect(todoItem).toBeInTheDocument();
    const deleteButton = screen.getByText('삭제');
    expect(deleteButton).toBeInTheDocument();

    const toDoList = screen.getByTestId('toDoList');
    expect(toDoList.childElementCount).toBe(1);

    fireEvent.change(input, { target: { value: 'study react 2' } });
    fireEvent.click(button);

    const todoItem2 = screen.getByText('study react 2');
    expect(todoItem2).toBeInTheDocument();
    expect(toDoList.childElementCount).toBe(2);

    const deleteButtons = screen.getAllByText('삭제');
    fireEvent.click(deleteButtons[0]);

    expect(todoItem).not.toBeInTheDocument();
    expect(toDoList.childElementCount).toBe(1);
  });

  it('로컬스토리지 데이터 투두리스트와 연동 확인', () => {
    localStorage.setItem(
      'TodoList',
      JSON.stringify(['todo 1', 'todo 2', 'todo 3'])
    );

    render(<App />);

    expect(screen.getByText('todo 1')).toBeInTheDocument();
    expect(screen.getByText('todo 2')).toBeInTheDocument();
    expect(screen.getByText('todo 3')).toBeInTheDocument();
    expect(screen.getAllByText('삭제').length).toBe(3);
  });
});
