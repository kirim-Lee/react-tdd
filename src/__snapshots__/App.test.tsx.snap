// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> 렌더링이 잘 되는지 테스트 1`] = `
<div>
  <div
    class="sc-bcXHqe iAIzKk"
  >
    <div
      class="sc-gswNZR qvSGk"
    >
      <div
        class="sc-jrcTuL inrruT"
        data-testid="toDoList"
      />
      <div
        class="sc-pyfCe eKYSuT"
      >
        <input
          class="sc-gKPRtg dqyRml"
          placeholder="할 일을 입력해 주세요"
          value=""
        />
        <button
          class="sc-eDvSVe cNqITy"
        >
          <div
            class="sc-jSUZER jTTUoN"
          >
            추가
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;
